python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -199,6 +222,32 @@ def get_partition_keys(self) -> Sequence[str]:
                 ]
         return []
 
+    def get_required_resource_keys(
+        self, node_def_snap: Union[CompositeSolidDefSnap, SolidDefSnap]
+    ) -> Sequence[str]:
+        all_keys = self.get_required_resource_keys_rec(node_def_snap)
+        return list(set(all_keys))
+
+    def get_required_resource_keys_rec(
+        self, node_def_snap: Union[CompositeSolidDefSnap, SolidDefSnap]
+    ) -> Sequence[str]:
+        if isinstance(node_def_snap, CompositeSolidDefSnap):
+            constituent_node_names = [
+                inv.solid_def_name
+                for inv in node_def_snap.dep_structure_snapshot.solid_invocation_snaps
+            ]
+            external_pipeline = self.get_external_pipeline()
+            constituent_resource_key_sets = [
+                self.get_required_resources(external_pipeline.get_node_def_snap(name))
```

Comment:
The Dagit updates look good to me! Will defer to @prha on the GraphQL changes.

I think one solution to the "does this required resource require config" challenge might be to expose more than the resource key here? If we were to have `GrapheneResourceRequirement(resource)` here instead of just the key set, we might be able to pull other fields off the resource requirement and get it's required info? It looks like this is calling through into code outside this PR though, so that may not be possible.

I think you're right that without that, we'd need two a follow-up query for more information about the resources.